### PR TITLE
Allow custom php.ini in embedded app

### LINKED ISSUE #472: php.ini in self-contained binary
Is it possible to embed custom php.ini configuration (such as short_open_tags) in my applications binary, or is it just best practice to keep the php.ini file somewhere on the server that is running the binary?

### PR DESCRIPTION
**Pull request by pierredup**
This adds the embedded path to the `PHP_INI_SCAN_DIR` env variable, so that a custom php.ini can be added to the root of a project.

Fixes #472 and #496

Screenshot 2024-01-22 at 13 55 21

### CODE REVIEW

File: caddy/php-server.go [hunk 1]
```diff
@@ -93,6 +94,10 @@ func cmdPHPServer(fs caddycmd.Flags) (int, error) {
 		} else if filepath.IsLocal(root) {
 			root = filepath.Join(frankenphp.EmbeddedAppPath, root)
 		}
+
+		if err := os.Setenv("PHP_INI_SCAN_DIR", ":"+frankenphp.EmbeddedAppPath); err != nil {
```
Comment (dunglas) [inline caddy/php-server.go:98]: I propose to change the env var only if:

* it isn't already set
* the embedded app contains a `php.ini` file

WDYT?

**Review by dunglas — CHANGES_REQUESTED** [review]

File: caddy/php-server.go (hunk 1 above)
Comment (pierredup) [reply caddy/php-server.go:98]: What about appending the embedded path if the `PHP_INI_SCAN_DIR` already exists? I.E if the `PHP_INI_SCAN_DIR` env variable exists and the embedded path contains a php.ini, then we append the embedded path to the existing `PHP_INI_SCAN_DIR` value?

Comment (dunglas) [reply caddy/php-server.go]: This works for me, as long as we change the variable only if a `php.ini` file is present at the root of the app.

**Review by dunglas — APPROVED** [review]

**Comment by dunglas** [comment]
Thank you!

**Pull request merged**